packages/next/server/api-utils.ts

```diff
@@ -63,6 +63,26 @@ export async function apiResolver(
       )
     }
 
+    let contentLength = 0
+    const writeData = apiRes.write
+    const endResponse = apiRes.end
+    apiRes.write = (...args: any[2]) => {
+      contentLength += Buffer.byteLength(args[0])
+      return writeData.apply(apiRes, args)
+    }
+    apiRes.end = (...args: any[2]) => {
+      if (args.length && typeof args[0] !== 'function') {
+        contentLength += Buffer.byteLength(args[0])
+      }
+
+      if (contentLength >= 5 * 1024 * 1024) {
+        console.warn(
+          `API response for ${req.url} exceeds 5mb. This will cause the request to fail in a future version.`
```

Comment:
This should have a message url linked to the error. I noticed that we don't have this documented so I've added docs for it and made it part of the PR template 👍 https://github.com/vercel/next.js/pull/26865/files#diff-0b82d9a8fc9c94f7729e7685fca44dd89a479746f7cb14171e6dcdc7beb94387R17

Reply:
Added 👍